src/GafferImage/ImageReader.cpp

```diff
@@ -292,15 +304,24 @@ void ImageReader::compute( ValuePlug *output, const Context *context ) const
 {
 	if( output == intermediateColorSpacePlug() )
 	{
-		std::string intermediateSpace = "";
+		std::string userColorSpace = colorSpacePlug()->getValue();
```

Comment:
I think the logic here can be simplified (and it needs to be changed somehow because all the tests are failing at present). My suggestion would be that if `colorSpacePlug->getValue() != ""`, we just use that as the result immediately, without even checking that it is a valid colorspace. If the user has entered an invalid space we need to alert them with an error (and the ColorSpace node should do this for us) rather than silently change behaviour.